src/components/ComponentPages/News/Edit/index.tsx

```diff
@@ -1,154 +1,139 @@
-      camp_num: +router.query?.camp[1]?.split("-")[0],
-      id: values.data.map((id) => id.id),
-      display_text: values.data.map((text) => text.display_text),
-      link: values.data.map((link) => link.link),
-      available_for_child: values.data.map(
-        (available) => available.available_for_child
-      ),
-    };
-    const res = await updateNewsDataApi(dataObj);
+    const res = await updateNewsFeedApi({
+      newsfeed_id: dataToUpdate?.id,
+      display_text: values.display_text,
+      link: values.link,
+      available_for_child: values.available_for_child,
+      submitter_nick_id: dataToUpdate?.submitter_nick_id,
+    });
     if (res?.status_code == 200) {
       router.back();
       return;
-    } else if (res?.status_code === 200) {
-      let noOfErr = Object.keys(res.error).map((err) => {
-        let err_on = err.replace("link.", "");
-        return err_on;
-      });
-      let urlError_dup = [...urlError];
-      noOfErr.forEach((val) => (urlError_dup[val] = !urlError_dup[val]));
-      setUrlError(urlError_dup);
-      setUrlErrorMsg(res.message);
-      setLoading(false);
+    } else if (res?.status_code == 400) {
+      setUrlError(true);
+      setUrlErrorMsg(res?.error?.link && res?.error?.link[0]);
     }
+    setLoading(false);
   };
-  if (dataToUpdate.length == 0) {
-    return null;
-  }
+
+  useEffect(() => {
+    async function getCampEditNewsDataCall() {
+      const reqBody = { newsfeed_id: +router.query?.camp[2]?.split("-")[0] };
+      const res = await getEditCampNewsFeedApi(reqBody);
+      const news = (res && res[0]) || {};
+      setDataToUpdate(news);
+      form.setFieldsValue({
+        display_text: news?.display_text,
+        link: news?.link,
+        available_for_child: news?.available_for_child,
+      });
+      if (isLogin === false) {
+        router.push("/login");
+      } else if (Object.keys(news).length === 0 && isLogin === true) {
+        router.push(`/topic/${router.query.camp[0]}/${router.query.camp[0]}`);
+      }
+    }
+    getCampEditNewsDataCall();
+  }, []);
+
+  if (isLogin === false) return null;
   return (
     <Card title="Edit News" className={styles.card}>
-      <Form
-        form={form}
-        name="basic"
-        initialValues={{
-          data: dataToUpdate,
-        }}
-        onFinish={onFinish}
-        autoComplete="off"
-        layout="vertical"
-      >
-        <Form.List name="data">
-          {(fields) => {
-            return (
-              <>
-                {fields.map((field, index) => (
-                  <>
-                    <Card
-                      className={styles.newsFormFieldsCard}
-                      bordered={false}
-                      key={field.key}
-                    >
-                      <Badge className={styles.newsFormFieldsCardCounter}>
-                        {field.key}
-                      </Badge>
+      <Form form={form} name="basic" layout={"vertical"} onFinish={onFinish}>
```

Comment:
I feel we can combine the form of Add News and Edit News. Both almost have the same fields

Reply:
refactored